### PR TITLE
FIX Remove workaround for nested redux form.

### PR DESCRIPTION
**Pull request by GuySartorelli**
Requires https://github.com/silverstripe/silverstripe-admin/pull/1694 and https://github.com/silverstripe/silverstripe-behat-extension/pull/266

## Issue
- https://github.com/silverstripe/silverstripe-admin/issues/1662

### CODE REVIEW

File: behat.yml [hunk 1]
```diff
@@ -15,9 +15,9 @@ default:
         - SilverStripe\Framework\Tests\Behaviour\CmsUiContext
         - SilverStripe\BehatExtension\Context\BasicContext
         - SilverStripe\BehatExtension\Context\LoginContext
-        - 
+        -
           SilverStripe\BehatExtension\Context\FixtureContext:
-            - '%paths.modules.linkfield%/tests/behat/features/files/'
+            - '%paths.modules.linkfield%/tests/behat/files/'
```
Comment (ChloeSartorelli) [inline behat.yml:20]: Unrelated refactoring - matches asset-admin. Files are not features, so they don't go in the features directory.

File: tests/behat/features/linkfield-in-elemental.feature [hunk 2]
```diff
@@ -51,7 +51,7 @@ I want to be able to work with LinkField and MultiLinkField in Elemental Block
     And I press the "Create link" button
     And I wait for 2 seconds
 
-    # Create ExternalLink in MultiLinkField
+    # Create PhoneLink in MultiLinkField
```
Comment (ChloeSartorelli) [inline tests/behat/features/linkfield-in-elemental.feature:54]: Unrelated refactoring - this looks like it was a copy/paste error.

File: tests/behat/features/create-edit-linkfield.feature
Comment (ChloeSartorelli) [inline tests/behat/features/create-edit-linkfield.feature:1]: I've refactored this slightly to make it easier to test individual bits locally. There's no sense waiting for all the "the dropdown shows these items" tests to pass when all you want to test is link creation.

File: tests/behat/features/create-edit-linkfield.feature [hunk 3]
```diff
@@ -5,141 +5,204 @@ I want to add links to pages, files, external URLs, email addresses and phone nu
...
-      And the "group" "EDITOR" has permissions "Access to 'Pages' section"
-      And I am logged in as a member of "EDITOR" group
-      And I go to "/admin/pages"
-      And I should see "Link Page"
-      And I click on "Link Page" in the tree
-
-  Scenario: I click on the link fields and see the list of allowed link types with icons for LinkFields
+    And I go to "/dev/build?flush"
+    And a "page" "Link Page"
+    And a "image" "folder1/file1.jpg"
```
Comment (ChloeSartorelli) [inline tests/behat/features/create-edit-linkfield.feature:10]: Adding this file is new

File: tests/behat/features/create-edit-linkfield.feature [hunk 4]
```diff
@@ -5,141 +5,204 @@ I want to add links to pages, files, external URLs, email addresses and phone nu
...
+    When I fill in "LinkText" with "File link"
+    And I click "Choose existing" in the ".uploadfield" element
+    And I press the "Back" HTML field button
+    # open "folder1"
+    And I click on the ".gallery__folders label[for='item-1']" element
+    # select "file1"
+    And I click on the ".gallery__files .gallery-item[role='button']" element
+    And I press the "Insert" button
+    And I press the "Create link" button
+    And I wait for 2 seconds
```
Comment (ChloeSartorelli) [inline tests/behat/features/create-edit-linkfield.feature]: Creating these two link types is new - wasn't being tested before

File: tests/behat/features/create-edit-linkfield.feature [hunk 5]
```diff
@@ -5,141 +5,204 @@ I want to add links to pages, files, external URLs, email addresses and phone nu
...
+    # Resize screen so we have "insert file" instead of just "insert" - they're actually buttons for completely different forms.
+    And I set the screen width to 700px
+    And I press the "Insert file" button
+    And I press the "Create link" button
+    And I reset the screen size
+    And I wait for 2 seconds
+
+    And I should see "Link to a file" in the "[data-field-id='Form_EditForm_HasManyLinks'] .link-picker__link--is-first" element
+    And I should see "folder1/file1.jpg" in the "[data-field-id='Form_EditForm_HasManyLinks'] .link-picker__link--is-first" element
+    And I should see "Draft" in the "[data-field-id='Form_EditForm_HasManyLinks'] .link-picker__link--is-first" element
```
Comment (ChloeSartorelli) [inline tests/behat/features/create-edit-linkfield.feature]: This test is new - it validates that the fix for nested redux forms works.

**Review by sabina-talipova — APPROVED** [review]
LGTM. Tested locally.

**Comment by sabina-talipova** [comment]
@GuySartorelli, I've approved since I tested locally all PRs together and tests were passed. But there is a small issue in CI. Installed version of `silverstripe/behat-extension` is `5.1.0` instead of `5.x-dev`([See here](https://github.com/silverstripe/silverstripe-linkfield/actions/runs/8058835376/job/22151577569?pr=239#step:9:535)). Is it possible to fix this issue by adding new behat methods in `5.1` or somehow else?

**Comment by GuySartorelli** [comment]
Tagged [5.2.0](https://github.com/silverstripe/silverstripe-behat-extension/releases/tag/5.2.0) of behat-extensions, rerunning CI

File: tests/behat/features/create-edit-linkfield.feature [hunk 6]
```diff
@@ -5,141 +5,203 @@ I want to add links to pages, files, external URLs, email addresses and phone nu
...
+    Then I should see "Link to a file" in the "[data-field-id='Form_EditForm_HasManyLinks'] .dropdown-item:nth-of-type(2)" element
+    When I click on the "[data-field-id='Form_EditForm_HasManyLinks'] .dropdown-item:nth-of-type(2)" element
+    And I wait for 5 seconds
+    Then I should see "Link to a file" in the ".modal-header" element
+    And I should see "Open in new window" in the ".modal-content" element
+    When I fill in "LinkText" with "File link"
+    And I click "Choose existing" in the ".uploadfield" element
+    And I press the "Back" HTML field button
+    # open "folder1"
+    And I click on the ".gallery__folders label[for='item-1']" element
```
Comment (ChloeSartorelli) [inline tests/behat/features/create-edit-linkfield.feature]: It turns out "item-1" isn't the first item, it's the item with ID=1 (I think you tried to tell me that Sabina, sorry I didn't understand at the time).
Not sure why the separate behat runs had different IDs but either way relying on IDs is bad.

To resolve this, I need to get the nth (in this case 1st) child of `.gallery__folders` (which will be the nth folder's holder element), and then get the label that is a child of that element.

I'll make that change now.

Comment (ChloeSartorelli) [reply tests/behat/features/create-edit-linkfield.feature]: Done - behat should pass now.
